### DISCUSSION TITLE
Type error

### DISCUSSION THREAD

**Discussion by alberto-bueno** [IDAES/idaes-pse#1435] [Q&A] [answered]
Dear all,

I just installed Idaes version 2.5.0 in a new Anaconda environment using the commands from the Windows Installation Guide.

All the installation runs without error except when I run: pytest --pyargs idaes -W ignore

I get the following error in more then 1000 times: **TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases**

When running any example from github, for example the TSA i get:
`
m = ConcreteModel() 
m.fs = FlowsheetBlock(dynamic=False)

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[7], line 5
      2 m = ConcreteModel()
      4 # create flowsheet
----> 5 m.fs = FlowsheetBlock(dynamic=False)

File ~\Anaconda3\envs\IDAES\envs\my-idaes-env\lib\site-packages\idaes\core\base\process_block.py:165, in ProcessBlock.__new__(cls, *args, **kwds)
    163 if args == ():  # no args so make scalar class
    164     bname = "_Scalar{}".format(cls.__name__)
--> 165     n = _ScalarProcessBlockMeta(bname, (cls._ComponentDataClass, cls), {})
    166     return n.__new__(n)  # calls this __new__() again with scalar class
    167 else:  # args so make indexed class

File ~\Anaconda3\envs\IDAES\envs\my-idaes-env\lib\site-packages\idaes\core\base\process_block.py:133, in _ScalarProcessBlockMeta.__new__(mcs, name, bases, dct)
    131 # provide function ``base_class_module()`` to get unit module, for visualizer
    132 dct["base_class_module"] = lambda mcs: bases[0].__module__
--> 133 return type.__new__(mcs, name, bases, dct)

TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases

`
Does anyone has any idea why is this happening and how can I solve it?

**Comment by andrewlee94** [comment]
@alberto-bueno This issue was caused by some changes Pyomo made in their last release, however these should all have been fixed in the IDAES v2.5 release. Could you double check:

1. the version of Pyomo that is installed
2. the version of IDAES that is installed and how you installed it (pip install, clone, etc.).

If you could prove a full list of your installed dependencies it might also help (I think `pip list` will provide this).

**Comment by lbianchi-lbl** [comment, marked as answer]
@alberto-bueno piggy-backing on @andrewlee94's comment:

- I believe this error is due to the fact that IDAES 2.4.0 (where this exact error is know to occur; see #1422), and not 2.5.0, is the version installed of your system
- This is because 2.5.0 (which was released within the last two weeks) doesn't yet have a Conda package, so the latest version that's available when installing using `conda install` is 2.4.0
- We're in the process of changing our Conda packaging process, so it might take a while for 2.5.0 to be available for Conda
- Until then, 2.5.0 can be installed with `pip install` following e.g. the steps below:

```sh
conda create --name idaes-pse -c conda-forge --yes python=3.10
conda activate idaes-pse
pip install idaes-pse==2.5.0
pip show idaes-pse pyomo  # should show idaes-pse 2.5.0, pyomo 6.7.3
idaes get-extensions --verbose
pytest --pyargs idaes -W ignore  # the TypeError shouldn't appear
```
